nucypher/policy/conditions/lingo.py

```diff
@@ -443,3 +560,46 @@ def check_version_compatibility(cls, version: str):
             raise InvalidConditionLingo(
                 f"Version provided, {version}, is incompatible with current version {cls.VERSION}"
             )
+
+
+class BaseExecAccessControlCondition(AccessControlCondition):
+    class Schema(AccessControlCondition.Schema):
+        return_value_test = fields.Nested(
+            ReturnValueTest.ReturnValueTestSchema(), required=True
+        )
+
+    def __init__(
+        self,
+        condition_type: str,
+        return_value_test: ReturnValueTest,
+        name: Optional[str] = None,
+        *args,
+        **kwargs,
+    ):
+        self.return_value_test = return_value_test
+        try:
+            self.execution_call = self._create_execution_call(*args, **kwargs)
+        except ValueError as e:
+            raise InvalidCondition(str(e))
+
+        super().__init__(condition_type=condition_type, name=name)
+
+    @abstractmethod
+    def _create_execution_call(self, *args, **kwargs) -> ExecutionCall:
+        """
+        Returns the execution call that the condition executes.
```

Comment:
I guess we can regard an `ExecutionCall` condition as being conditioned on the call returning something that can be processed by the next condition? Or is there not actually a condition here – i.e. whatever is returned, the sequence continues? 

e.g. developer creates an `ExecutionCall` to call a weather API and get today's temperature, which should then be fed into the next `Condition`. If an error is returned rather than an integer – will that still be fed into the `Condition`? Would the developer have to explicitly add a `ReturnValueTest` to the first `ExecutionCall` to avoid that happening?

Reply:
For now, SequentialCondition only uses Conditions (not `ExecutionCalls`) which mandate the use of a `ReturnValueTest`.

`ExecutionCall`, for now, is purely an internal low-level abstraction of the execution that a condition does before any condition is checked. Currently, developers are not exposed to calls at all, only conditions i.e. same as before.

For example, for an RPCCondition, it uses an `RPCCall` object and calls `execute_call.execute()` which simply calls `get_balance(...)` and returns the result. The RPCCondition takes this result and then performs the actual condition check using its ReturnValueTest. 

This repeats i.e. TimeCondition uses TimeRPCCall, ContractCondition uses a ContractCall etc.

So all conditions that involve execution (i.e. not logical calls) follow the same internal paradigm:

```python
class Condition:
    def __init__(...):
        self.execution_call = ...
        self.return_value_test = ...
        
    def verify(...):
        execution_result = self.execution_call.execute().  # the on-chain method to execute
        condition_check_success = self.return_value_test.eval(execution_result).  # check whether result passes the condition check i.e. True/False
        return condition_check_success, execution_result.  # return tuple of ("condition successful", "execution result")
}
```

The returned `condition_check_success` is used to check whether the condition passed or not i.e `True`/`False`

For SequentialConditions if any condition does not pass i.e. the `condition_check_success` value is `False` or the execution raised an exception (eg. ConnectionError connecting to a json web server), no subsequent condition is executed/checked and the overall condition fails. If that condition passes, then the `execution_result` is stored as the `varName` for the ConditionVariable for that condition and subsequent conditions can use `:<varName>` to use the returned value in their execution/return value check.

---

It is possible that based on discussions or feedback, we could decide that SequentialConditions should allow for any combination of Condition and ExecutionCalls so that devs don't always have to provide `ReturnValueTest` if they don't need any sanity checking of values before they are used. This was explored earlier in the PR when I initially mused on it. See 2nd bullet point in https://github.com/nucypher/nucypher/issues/3555.